src/Config.ts

```diff
@@ -1,21 +1,24 @@
+export type ExposeOption = "all" | "none" | "export";
+export type JSDocOption = "none" | "extended" | "basic";
+
 export interface Config {
     path?: string;
     type?: string;
     tsconfig?: string;
-    expose: "all" | "none" | "export";
-    topRef: boolean;
-    jsDoc: "none" | "extended" | "basic";
+    expose?: ExposeOption;
+    topRef?: boolean;
+    jsDoc?: JSDocOption;
     sortProps?: boolean;
     strictTuples?: boolean;
     skipTypeCheck?: boolean;
     encodeRefs?: boolean;
     extraTags?: string[];
 }
 
-export const DEFAULT_CONFIG: Config = {
-    expose: "export",
+export const DEFAULT_CONFIG = {
```

Comment:
I think this should still be of type `Config`, no?

Reply:
Yea, it seems like that shoudl work, but unforunately it doesn't.

When it's declared of type Config, TypeScript then believes `{ ...DEFAULT_CONFIG, ... config }` is a Config and so the optionals are still optional. By removing the type annotation, TypeScript correctly infers that the default values being provided here are present, so it know `mergedConfig` has these values present, too.

Reply:
So do we want to say `Required<Config>` instead?

Reply:
I did not know that was a thing! But it doesn't seem to work, because `Required<Config>` doesn't declare `path`, `type`, or `tsconfig`. I tried `Partial<Required<Config>>` too which works here, but has the same problem which is it doesn't know e.g. `expose` is defined on `mergedConfig`.

I could create a second type, though, maybe using `Omit` or one of the related operators.

Reply:
Yeah, let's use `Omit` then. I'd like to avoid `as` as much as possible.